### ISSUE TITLE
multiple mutablePut failure (maybe my misunderstanding?)

### ISSUE THREAD

**Issue by robinsloan** [hyperswarm/dht#63]
I am new to this software, & really enjoying it. I can get a bootstrapped private DHT up and running, start servers & connect to them, `mutablePut` a value and then `mutableGet`  that value back out. It's really cool!

However, when I try to `mutablePut` a new value onto the same key, my bootstrap nodes error out with this message:

```
/home/robin/dht-fun/node_modules/compact-encoding/index.js:280
      state.buffer.set(s, state.start)
                   ^

TypeError: Cannot convert undefined or null to object
    at Buffer.set (<anonymous>)
    at Object.encode (/home/robin/dht-fun/node_modules/compact-encoding/index.js:280:20)
    at Object.encode (/home/robin/dht-fun/node_modules/@hyperswarm/dht/lib/messages.js:269:15)
    at Object.encode (/home/robin/dht-fun/node_modules/compact-encoding/index.js:373:7)
    at Persistent.onmutableput (/home/robin/dht-fun/node_modules/@hyperswarm/dht/lib/persistent.js:186:26)
    at HyperDHT.onrequest (/home/robin/dht-fun/node_modules/@hyperswarm/dht/index.js:278:26)
    at HyperDHT._onrequest (/home/robin/dht-fun/node_modules/dht-rpc/index.js:399:14)
    at IO.onmessage (/home/robin/dht-fun/node_modules/dht-rpc/lib/io.js:53:12)
    at Socket.emit (node:events:390:28)
    at UDP.onMessage [as onmessage] (node:dgram:939:8)
```

That's triggered by this code from a client node, in which I attempt to `mutablePut` again onto a public key that's previously been `mutablePut` onto:

```
const mutableKeyPair = DHT.keyPair(Buffer.alloc(32).fill('mutability'))
const put = await node.mutablePut(mutableKeyPair, Buffer.from('a new value!'))
```

Note that the error throws on the bootstrap nodes, not the client node making this request. I suspect this is a silly/simple mistake or misconception on my part… which is why I came here to see if anyone has any clues. Thank you!

**Comment by mafintosh** [comment]
Thanks! It was a silly typo that the tests didn't catch (embarassing), I just fixed it in 5.0.8

**Comment by robinsloan** [comment]
Ah, I'm glad to have helped spot a little bug! Thanks for this fix.
